### PR TITLE
Switched to single activity architecture using Navigation component

### LINKED ISSUE #33: Refactor app to single activity architecture

### PR DESCRIPTION
**Pull request by zukovicfilip**
This fix #33 
I used Navigation component to switch app to single activity architecture.
Converted all DialogFragmments and Activities to Fragments.

### CODE REVIEW

File: app/src/main/res/values/facebook_api.xml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<resources>
```
Comment (SlobodanPrijic) [inline app/src/main/res/values/facebook_api.xml]: @zukovicfilip please do not add these files to Git.

File: app/src/main/java/com/hyperether/getgoing/ui/fragment/TrackingFragment.java [hunk 2]
```diff
@@ -401,13 +430,14 @@ public void onClick(DialogInterface paramDialogInterface, int paramInt) {
...
+    private final View.OnClickListener mButtonBackListener = v -> getActivity().onBackPressed(); //onAnyBackButtonPressed();
 
-    private final OnClickListener mButtonSetGoalListener = new OnClickListener() {
+    private final View.OnClickListener mButtonSetGoalListener = new View.OnClickListener() {
         @Override
         public void onClick(View v) {
-            ActivitiesFragment activitiesFragment = ActivitiesFragment.newInstance(null, OPENED_FROM_LOCATION_ACT);
-            activitiesFragment.show(getSupportFragmentManager(), "ActivitiesFragment");
+            Bundle bundle = new Bundle();
+            bundle.putInt("from", OPENED_FROM_LOCATION_ACT);
```
Comment (SlobodanPrijic) [inline app/src/main/java/com/hyperether/getgoing/ui/fragment/TrackingFragment.java]: @zukovicfilip do not hardcode parameters ("from"). Please create constants / strings.

File: app/src/main/java/com/hyperether/getgoing/ui/fragment/TrackingFragment.java [hunk 3]
```diff
@@ -109,32 +116,51 @@
...
         mRouteAlreadySaved = true;
 
         // Open the shared preferences
-        mPrefs = getSharedPreferences("SharedPreferences", Context.MODE_PRIVATE);
+        mPrefs = getActivity().getSharedPreferences("SharedPreferences", Context.MODE_PRIVATE);
         // Get a SharedPreferences editor
         mEditor = mPrefs.edit();
         // Get user data shared prefs
-        cbPrefs = getSharedPreferences("CBUserDataPref.txt", Context.MODE_PRIVATE);
+        cbPrefs = getActivity().getSharedPreferences("CBUserDataPref.txt", Context.MODE_PRIVATE);
```
Comment (SlobodanPrijic) [inline app/src/main/java/com/hyperether/getgoing/ui/fragment/TrackingFragment.java]: @zukovicfilip why do we need activity to access shared preferences? Context should be enough.

File: app/src/main/java/com/hyperether/getgoing/ui/fragment/ShowDataFragment.java [hunk 4]
```diff
@@ -122,7 +115,7 @@ public void onDestroy() {
     }
 
     private void initializeViewModel() {
-        routeViewModel = ViewModelProviders.of(getActivity()).get(RouteViewModel.class);
+        routeViewModel = new ViewModelProvider(this).get(RouteViewModel.class);
```
Comment (SlobodanPrijic) [inline app/src/main/java/com/hyperether/getgoing/ui/fragment/ShowDataFragment.java:119]: @zukovicfilip is this only view which access to RouteViewModel?

File: app/src/main/java/com/hyperether/getgoing/ui/adapter/DbRecyclerAdapter.java [hunk 5]
```diff
@@ -26,7 +27,7 @@
     private GgOnClickListener listener;
 
 
-    public DbRecyclerAdapter(Context context, List<DbRoute> routes) {
+    public DbRecyclerAdapter(Fragment context, List<DbRoute> routes) {
```
Comment (SlobodanPrijic) [inline app/src/main/java/com/hyperether/getgoing/ui/adapter/DbRecyclerAdapter.java]: @zukovicfilip Fragment instance variable should not be called "context"

**Review by SlobodanPrijic — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/hyperether/getgoing/ui/fragment/ShowDataFragment.java (hunk 4 above)
Comment (zukovicfilip) [reply app/src/main/java/com/hyperether/getgoing/ui/fragment/ShowDataFragment.java:119]: Yes

**Review by SlobodanPrijic — APPROVED** [review]

**Pull request merged**